web_navbar_chameleon/models/res_company.py

```diff
@@ -0,0 +1,164 @@
+            self.navbar_color_text = ''
+
+    @api.onchange('navbar_color_background', 'navbar_color_background_hover',
+                  'navbar_color_text')
+    def change_navbar_color(self):
+        if self._origin.id:
+            datas = base64.b64encode(
+                self._generate_navbar_chameleon_scss().encode("utf-8"))
+            self._create_or_update_navbar_chameleon_scss_attachment(
+                self._origin.id, datas)
+
+    @api.multi
+    def _generate_navbar_chameleon_scss(self):
+        self.ensure_one()
+
+        return """
+            .o_main_navbar {
+              background-color: %(bg_color)s !important;
+              color: %(text_color)s !important;
+
+              > .o_menu_brand {
+                color: %(text_color)s !important;
+                &:hover, &:focus, &:active, &:focus:active {
+                  background-color: %(bghover_color)s !important;
+                }
+              }
+
+              .show {
+                .dropdown-toggle {
+                  background-color: %(bghover_color)s !important;
+                }
+              }
+
+              > ul {
+                > li {
+                  > a, > label {
+                    color: %(text_color)s !important;
+
+                    &:hover, &:focus, &:active, &:focus:active {
+                      background-color: %(bghover_color)s !important;
+                    }
+                  }
+                }
+              }
+            }
+        """ % {
+            'bg_color': self.navbar_color_background or '$o-brand-odoo',
+            'bghover_color': self.navbar_color_background_hover
+            or '$o-navbar-inverse-link-hover-bg',
+            'text_color': self.navbar_color_text or '#fff',
+        }
+
+    @api.model
+    def _create_or_update_navbar_chameleon_scss_attachment(self, company_id,
+                                                           datas):
+        custom_url = URL_SCSS_GEN % company_id
+        IrAttachmentObj = self.env["ir.attachment"]
+        custom_attachment = IrAttachmentObj.search([("url", '=', custom_url)])
+        if custom_attachment:
+            custom_attachment.write({"datas": datas})
+        else:
+            IrAttachmentObj.create({
+                'name': custom_url,
+                'type': "binary",
+                'mimetype': "text/scss",
+                'datas': datas,
+                'datas_fname': custom_url.split("/")[-1],
+                'url': custom_url,
+            })
+
+        self.env["ir.qweb"].clear_caches()
+
+    @api.multi
+    def update_navbar_chameleon_scss(self):
+        for record in self:
+            datas = base64.b64encode(
+                record._generate_navbar_chameleon_scss().encode("utf-8"))
+            self._create_or_update_navbar_chameleon_scss_attachment(record.id,
+                                                                    datas)
+
+    def _get_image_mix_color_rgb(self, img64):
```

Comment:
same here: I'd move it to a global func. 
You can collect them all in a `utils.py` file.